ts/src/hashkey.ts

```diff
@@ -0,0 +1,4313 @@
+            params['newClientOrderId'] = clientOrderId;
+        }
+        return this.extend (request, params);
+    }
+
+    createSwapOrderRequest (symbol: string, type: OrderType, side: OrderSide, amount: number, price: Num = undefined, params = {}): Dict {
+        /**
+         * @method
+         * @ignore
+         * @name hashkey#createSwapOrderRequest
+         * @description helper function to build request
+         * @param {string} symbol unified symbol of the market to create an order in
+         * @param {string} type 'market' or 'limit' or 'STOP'
+         * @param {string} side 'buy' or 'sell'
+         * @param {float} amount how much of you want to trade in units of the base currency
+         * @param {float} [price] the price that the order is to be fulfilled, in units of the quote currency, ignored in market orders
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {bool} [params.postOnly] if true, the order will only be posted to the order book and not executed immediately
+         * @param {bool} [params.reduceOnly] true or false whether the order is reduce only
+         * @param {float} [params.triggerPrice] The price at which a trigger order is triggered at
+         * @param {string} [params.timeInForce] 'GTC', 'FOK', 'IOC', 'LIMIT_MAKER' or 'PO'
+         * @param {string} [params.clientOrderId] a unique id for the order
+         * @returns {object} request to be sent to the exchange
+         */
+        const market = this.market (symbol);
+        const request: Dict = {
+            'symbol': market['id'],
+            'type': 'LIMIT',
+            'quantity': this.amountToPrecision (symbol, amount),
+        };
+        const isMarketOrder = type === 'market';
+        if (isMarketOrder) {
+            request['priceType'] = 'MARKET';
+        }
+        if (price !== undefined) {
+            request['price'] = this.priceToPrecision (symbol, price);
+            request['priceType'] = 'INPUT';
+        }
+        let reduceOnly = false;
+        [ reduceOnly, params ] = this.handleParamBool (params, 'reduceOnly', reduceOnly);
+        let suffix = '_OPEN';
+        if (reduceOnly) {
+            suffix = '_CLOSE';
+        }
+        request['side'] = side.toUpperCase () + suffix;
+        let timeInForce: Str = undefined;
+        [ timeInForce, params ] = this.handleParamString (params, 'timeInForce');
+        let postOnly = false;
+        [ postOnly, params ] = this.handlePostOnly (isMarketOrder, timeInForce === 'LIMIT_MAKER', params);
+        if (postOnly) {
+            timeInForce = 'LIMIT_MAKER';
+        }
+        if (timeInForce !== undefined) {
+            request['timeInForce'] = timeInForce;
+        }
+        const clientOrderId = this.safeString (params, 'clientOrderId');
+        if (clientOrderId === undefined) {
+            throw new ArgumentsRequired (this.id + ' createSwapOrder() requires a clientOrderId parameter');
```

Comment:
if the user does not provide it we should generate one using for instance this.uuid instead of throwing.


Besides that, since `createSwapOrder` is not a unified method, if we need to throw an exception, we should use `createOrder` as the method's name instead

Reply:
Thanks, will know!